test/unit/puppetlabs/services/jruby/jruby_puppet_service_test.clj

```diff
@@ -106,8 +105,16 @@
           (is (instance? JRubyPuppet jruby-puppet))
           (is (= 0 (jruby-protocol/free-instance-count service))))
         (is (= 1 (jruby-protocol/free-instance-count service)))
+        ;; borrow and return one more time
+        (with-jruby-puppet
+          jruby-puppet
+          service)
         (let [jruby (jruby-protocol/borrow-instance service)]
-          (is (= 2 (:request-count (jruby-core/instance-state jruby)))))))))
+          ;; the counter gets incremented when the instance is returned to the
+          ;; pool, so right now it should be at 2 since we've called
+          ;; `with-jruby-puppet` twice.
+          (is (= 2 (:request-count (jruby-core/instance-state jruby))))
```

Comment:
Perhaps rename this key to `:borrow-count` to maintain the separation of concerns between the JRuby service and request handling.